### PR TITLE
sanitycheck: enable valgrind checking using an option

### PR DESCRIPTION
**Pull request by nashif**
Enable valgrind using --valgrind option and fix reporting.

### CODE REVIEW

File: scripts/sanitycheck [hunk 1]
```diff
@@ -566,12 +567,14 @@ class BinaryHandler(Handler):
         else:
             command = [self.binary]
 
+        run_valgrid = False
```
Comment (pabigot) [inline scripts/sanitycheck]: s/run_valgrid/run_valgrind/g

**Review by vanwinkeljan — APPROVED** [review]

**Comment by aescolar** [comment]
Something amusing:
```sanitycheck -p native_posix  -T samples/hello_world/ --valgrind -vv```
produces:
```
1/1 native_posix              samples/hello_world/sample.helloworld              FAILED Valgrind error (native 0.650s)

	see: sanity-out/native_posix/samples/hello_world/sample.helloworld/handler.log
```
(meaning pointing to the handler.log)
while
```sanitycheck -p native_posix  -T samples/hello_world/ --valgrind ```

```
native_posix              samples/hello_world/sample.helloworld              FAILED: Valgrind error

	see: sanity-out/native_posix/samples/hello_world/sample.helloworld/valgrind.log
```
points to the right file

**Comment by nashif** [comment]
> Something amusing:

this is not fixed

**Comment by aescolar** [comment]
> this is now fixed

Kinda, with this latest fix running with `-v` print the log file twice:
```
sanitycheck -p native_posix  -T samples/hello_world/  -v
Renaming output directory to zephyr/sanity-out.17
JOBS: 16
Reading test case configuration files under zephyr/samples/hello_world...
Reading platform configuration files under zephyr/boards...
Reading platform configuration files under zephyr/scripts/sanity_chk/boards...
Building initial testcase list...
1 test configurations selected, 0 configurations discarded due to filters.
Adding tasks to the queue...

	see: sanity-out/native_posix/samples/hello_world/sample.helloworld/build.log
1/1 native_posix              samples/hello_world/sample.helloworld              FAILED Build failure (native)

	see: sanity-out/native_posix/samples/hello_world/sample.helloworld/build.log

0 of 1 tests passed (0.00%), 1 failed, 0 skipped with 0 warnings in 1.44 seconds
In total 1 test cases were executed on 1 out of total 208 platforms (0.48%)
```
Not a too bad issue though

**Comment by nashif** [comment]
> Kinda, with this latest fix running with `-v` print the log file twice:

yes, this code is annoying, I did not get to refactor it yet, the logic is mind-twisting.

File: scripts/sanitycheck [hunk 2]
```diff
@@ -3342,6 +3347,12 @@ structure in the main Zephyr tree: boards/<arch>/<board_name>/""")
         NOTE: west-flash must be enabled to use this option.
         """
     )
+    parser.add_argument(
+        "--valgrind", action="store_true",
+	help="""Run binary through valgrind and check for memory leaks.
```
Comment (aescolar) [inline scripts/sanitycheck]: Better rephrase a bit to:
"and check for memory leaks." => "and check for several memory access errors"

---------------------

Background: you won't catch Zephyr leaks with it, but you do catch other issues:
* Accessing memory you shouldn't, e.g. overrunning and underrunning heap blocks, overrunning the top of the stack, and accessing memory after it has been freed.
* Using undefined values, i.e. values that have not been initialised, or that have been derived from other undefined values.
* Overlapping src and dst pointers in memcpy and related functions.
* Passing a fishy (presumably negative) value to the size parameter of a memory allocation function.

File: scripts/sanitycheck [hunk 3]
```diff
@@ -522,7 +523,6 @@ class BinaryHandler(Handler):
         """
         super().__init__(instance, type_str)
 
-        self.valgrind = False
         self.terminated = False
```
Comment (aescolar) [inline scripts/sanitycheck]: You are leaving more dead code around :) (there is no use of this self.terminated anymore)
(just a remark, you don't necessarily need to do anything)

File: scripts/sanitycheck [hunk 4]
```diff
@@ -605,17 +607,22 @@ class BinaryHandler(Handler):
 
         subprocess.call(["stty", "sane"])
         self.instance.results = harness.tests
-        if not self.terminated and self.returncode != 0:
+
+        if self.returncode != 0:
             #When a process is killed, the default handler returns 128 + SIGTERM
```
Comment (aescolar) [inline scripts/sanitycheck]: This comment was meant to explain why the if above had the condition "if not self.terminated"
Now, this comment, even that it is informative, is a bit irrelevant (Meaning, you can delete it if you want)

Anyhow with the 60 seconds timeout is unlikely we will be terminating them for anything more than hung tests.

**Review by aescolar — APPROVED** [review]
Does what is promised. A few minor issues (but no need for me to re-review)

File: scripts/sanitycheck (hunk 3 above)
Comment (nashif) [reply scripts/sanitycheck]: I am still not done, as you can see, CI is still failing because of this change. Getting back to this now.

**Pull request merged**